src/julia.h

```diff
@@ -1492,6 +1495,8 @@ STATIC_INLINE void jl_eh_restore_state(jl_handler_t *eh)
     }
 #endif
     JL_SIGATOMIC_END();
+    jl_finalizers_enable(eh->finalizers_enabled);
+    jl_gc_enable(eh->gc_enabled);
```

Comment:
I think the new one is better in general. `gc_enable` really shouldn't be used in julia code so it's hard to say what's the expected usage pattern....

FYI, this adds another conditional safepoint to the beginning of the `catch` block =)